apps/homeless-people-help-map/src/components/App.tsx

```diff
@@ -280,19 +312,45 @@ export const App = () => {
         styles={DISTRICTS_STYLE}
       />
 
-      {terrainServices?.map(({ key, geojson }, index) => (
-        <Layer
-          key={index}
-          geojson={geojson}
-          // Style ids have to be unique
-          styles={(geojson.features[0]?.geometry.type === "Polygon"
-            ? TERRAIN_SERVICES_POLYGON_STYLE
-            : TERRAIN_SERVICES_POINT_STYLE
-          ).map((style) => ({ ...style, id: `${style.id}-${index}` }))}
-          isVisible={key === activeTerrainService?.key}
-          hoverPopup={Popup}
-        />
-      ))}
+      {terrainServices?.map(({ key, geojson }, index) => {
+        return (
+          <Layer
+            ignoreClick
+            key={index}
+            geojson={geojson}
+            // Style ids have to be unique
+            styles={(geojson.features[0]?.geometry.type === "Polygon"
+              ? TERRAIN_SERVICES_POLYGON_STYLE
+              : TERRAIN_SERVICES_POINT_STYLE
+            ).map((style) => ({ ...style, id: `terrainServices-${style.id}-${index}` }))}
+            isVisible={key === activeTerrainService?.key}
+            hoverPopup={Popup}
+          />
+        );
+      })}
+      {/* can't be all data be in one layer like it is in paas map? */}
```

Comment:
of course they can but that comes with other disadvantages. Tbh, give it a couple minutes of though on how would the logic look (not giving it as a "homework", I honestly don't know which parts it would affect besides this, but I have a feeling it would not be an isolated thing. But I may be wrong) and either a) update the comment to explanation b) update the comment to a TODO/FIXME and the steps needed for single layer or c) change it right away if you believe that's the best way forward